portality/scripts/remove_duplicate_articles.py

```diff
@@ -30,13 +39,19 @@
     snapshot = not args.ghost
     snapshot_report = "with snapshotting" if snapshot else "without snapshotting"
 
+    dry_run = args.dry_run
+
     start = datetime.now()
-    count = 0
-    print 'Starting {0} {snapshot} in 5 seconds, Ctrl+C to exit.'.format(start.isoformat(), snapshot=snapshot_report)
+
+    if dry_run:
+        print 'Starting {0} dry run in 5 seconds, Ctrl+C to exit.'.format(start.isoformat())
+    else:
+        print 'Starting {0} {snapshot} in 5 seconds, Ctrl+C to exit.'.format(start.isoformat(), snapshot=snapshot_report)
     time.sleep(5)
 
+    count = 0
     originals = []  # make sure we only delete FORWARD. So, if articles
-    # A, B and C are duplicates, we stumble upon A first, and delete B and C.
+    # A, B and C are duplicates, we stumble upon A first, and delete B and C.cat
     # We must then ensure we do not delete A if we chance upon B or C
     # (because they are still in Python memory even though they are not in ES).
     article_iterator = models.Article.iterall(page_size=5000)
```

Comment:
I seem to remember something about the order that things get considered.  I presume we want A to be the actual one to keep (e.g. the most recent, or the oldest, whatever the rule is).  iterall doesn't inherently order anything, I don't think, or if it does it is by ID.  So your ordering here is not guaranteed.

Reply:
This script has been completely rewritten in the part 2 pull request #1387 - it doesn't use an iterator any more, but a scroll. I should've really made this PR just to cover the changes in ```article.py```